defmt/src/lib.rs

```diff
@@ -361,11 +361,21 @@ pub use defmt_macros::timestamp;
 ///         const ABC = Self::A.bits | Self::B.bits | Self::C.bits;
 ///     }
 /// }
-///
+/// #[cfg(feature = "bitflagsv2")]
```

Comment:
I think we need some more documentation here to explain why we have two bitflags macros now and how users should pick between them. And actually, currently `bitflagsv2` is completely undocumented because the new docs were added to the block for `bitflags`. You can check with `cargo doc -p defmt --features=bitflagsv2 --open`.

<img width="995" alt="image" src="https://github.com/user-attachments/assets/69908943-50f2-47d7-839e-773e2eba83f9">

<img width="995" alt="image" src="https://github.com/user-attachments/assets/bc662358-be94-4327-999a-e73267b23458">